tests/system/src/test/java/test/robot/javafx/web/PointerEventTest.java

```diff
@@ -0,0 +1,225 @@
+        Util.runAndWait(() -> {
+            robot.mouseMove((int)(scene.getWindow().getX() + scene.getX() + DX),
+                            (int)(scene.getWindow().getY() + scene.getY() + DY));
+            robot.mousePress(buttons);
+        });
+
+        Util.runAndWait(() -> {
+            for (int i = 0; i < DRAG_DISTANCE; i++) {
+                // Move the mouse backwards so that the pointer does not stay on the popup, if any.
+                robot.mouseMove((int)(scene.getWindow().getX() + scene.getX() + DX - i),
+                                (int)(scene.getWindow().getY() + scene.getY() + DY));
+            }
+        });
+
+        Util.sleep(SLEEP_TIME);
+        Util.runAndWait(()  -> {
+            buttonMask = element.getTextContent();
+            robot.mouseRelease(buttons);
+        });
+
+        return Integer.parseInt(buttonMask);
+    }
+
+    @Test
+    public void testLeftButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.PRIMARY);
+        assertEquals(LEFT_BUTTON_DRAG, result);
+    }
+
+    @Test
+    public void testRightButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.SECONDARY);
+        assertEquals(RIGHT_BUTTON_DRAG, result);
+    }
+
+    @Test
+    public void testMiddleButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.MIDDLE);
+        assertEquals(MIDDLE_BUTTON_DRAG, result);
+    }
+
+    @Test
+    public void testLeftMiddleButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.PRIMARY, MouseButton.MIDDLE);
+        assertEquals((LEFT_BUTTON_DRAG | MIDDLE_BUTTON_DRAG), result);
+    }
+
+    @Test
+    public void testMiddleRightButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.MIDDLE, MouseButton.SECONDARY);
+        assertEquals((MIDDLE_BUTTON_DRAG | RIGHT_BUTTON_DRAG), result);
+    }
+
+    @Test
+    public void testLeftRightButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.PRIMARY, MouseButton.SECONDARY);
+        assertEquals((LEFT_BUTTON_DRAG | RIGHT_BUTTON_DRAG), result);
+    }
+
+    @Test
+    public void testLeftMiddleRightButtonDrag() {
+        int result = mouseButtonDrag(MouseButton.PRIMARY, MouseButton.MIDDLE, MouseButton.SECONDARY);
+        assertEquals((LEFT_BUTTON_DRAG | MIDDLE_BUTTON_DRAG | RIGHT_BUTTON_DRAG), result);
+    }
+
+    @BeforeClass
+    public static void initFX() {
+        startupLatch = new CountDownLatch(2);
+        new Thread(() -> Application.launch(TestApp.class, (String[])null)).start();
+        waitForLatch(startupLatch, 15, "Timeout waiting for FX runtime to start");
+    }
+
+    @AfterClass
+    public static void exit() {
+        Platform.runLater(() -> {
+            stage.hide();
+        });
+        Platform.exit();
+    }
+
+    @After
+    public void resetTest() {
+        Util.runAndWait(() -> {
+            robot.mouseRelease(MouseButton.PRIMARY, MouseButton.MIDDLE, MouseButton.SECONDARY);
+            robot.mouseClick(MouseButton.PRIMARY);
```

Comment:
I presume clicking the left mouse button is done to dismiss the popup that will happens when dragging with the right mouse button? This doesn't work on Windows, so subsequent tests fail. If you change it to pressing the `ESC` key, then it should work:

```
    robot.keyType(KeyCode.ESCAPE);
```

Reply:
Alternatively, moving the mouse up and/or to the left after releasing all the buttons and before clicking the middle button will work (on Windows the popup doesn't display until after you release the middle button, and is displayed at the then-current mouse position). Pressing the ESC seems easier, but it's up to you.

Reply:
Added code to press ESC key.Thanks for the sharing the observation.